src/check_ast.rs

```diff
@@ -372,10 +385,30 @@ where
                         }
                     }
                 }
+                if self.settings.select.contains(&CheckCode::E741) {
+                    self.checks
+                        .extend(checks::check_ambiguous_variable_name(target));
+                }
             }
             StmtKind::Delete { .. } => {
                 self.seen_non_import = true;
             }
+            StmtKind::For { target, .. } => {
+                if self.settings.select.contains(&CheckCode::E741) {
+                    self.checks
+                        .extend(checks::check_ambiguous_variable_name(target));
+                }
+            }
+            StmtKind::With { items, .. } | StmtKind::AsyncWith { items, .. } => {
+                if self.settings.select.contains(&CheckCode::E741) {
+                    for item in items {
+                        if let Some(vars) = &item.optional_vars {
+                            self.checks
+                                .extend(checks::check_ambiguous_variable_name(vars))
+                        }
```

Comment:
Do you have a sense for how many of these would be caught by instead hooking into `handle_node_store`? Could it help reduce the number of cases we have to handle?

Reply:
> Do you have a sense for how many of these would be caught by instead hooking into handle_node_store?

@charliermarsh Thanks for the comment. By `hooking into handle_node_store`, you meant something like below? 

```rust
    fn handle_node_store(&mut self, expr: &Expr, parent: Option<&Stmt>) {
        if let ExprKind::Name { id, .. } = &expr.node {
            ...

            // For example, `should_check_ambiguous_variable_name` should be True
            // for Name in Assign.targets
            let should_check_ambiguous_variable_name = ...;
            if self.settings.select.contains(&CheckCode::E741)
                && should_check_ambiguous_variable_name
                && checks::is_ambiguous_name(id)
            {
                self.checks.push(Check::new(
                    CheckKind::AmbiguousVariableName(id.to_string()),
                    expr.location,
                ));
            }
```

Reply:
Yeah that's roughly what I was suggesting!